File: packages/sdk-react-native/ios/Options.m
#import "Options.h"
#import <React/RCTLog.h>

@implementation Options

@synthesize dappName = _dappName;
@synthesize dappUrl = _dappUrl;
@synthesize dappIconUrl = _dappIconUrl;
@synthesize dappScheme = _dappScheme;
@synthesize infuraAPIKey = _infuraAPIKey;
@synthesize apiVersion = _apiVersion;


- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
  if (self = [super init]) {
    _dappName = dictionary[@"dappName"];
    _dappUrl = dictionary[@"dappUrl"];
    _dappIconUrl = dictionary[@"dappIconUrl"];
    _dappScheme = dictionary[@"dappScheme"];
    _infuraAPIKey = dictionary[@"infuraAPIKey"];
    _apiVersion = dictionary[@"apiVersion"];
  }
  return self;
}

@end

File: packages/sdk-react-native/ios/Options.swift
//
//  Metadata.swift
//  MetaMaskReactNativeSdk
//

import Foundation

@objc class Options: NSObject {
  let dappName: String
  let dappUrl: String
  let dappIconUrl: String
  let dappScheme: String
  let infuraAPIKey: String

  @objc init(dappName: String, dappUrl: String, dappIconUrl: String, dappScheme: String, infuraAPIKey: String) {
    self.dappName = dappName
    self.dappUrl = dappUrl
    self.dappIconUrl = dappIconUrl
    self.dappScheme = dappScheme
    self.infuraAPIKey = infuraAPIKey
    super.init()
  }
}
